### DISCUSSION TITLE
How to setup reporting in converter for new device?

### DISCUSSION THREAD

**Discussion by tagy** [Koenkk/zigbee2mqtt#12171] [Devices]
I'm working on a custom device that uses the nRF52 chip.

I have added a custom converter and the device can send an on/off commands that works correctly. But I want the device to be able to report the battery percentage remaining. The only way I can get this to work is through the front-end, by first adding reporting of the powerCfg->batteryRemainingPercentage. Once I have added that in the UI it works and the device can report the battery.

But how can I setup that reporting automatically, so I do not need to use the front-end UI?

I have tried with the following configure in my external converter, but I see the following error

` Failed to configure '0xf4ce36c170076c1e', attempt 1 (TypeError: Cannot read properties of undefined (reading 'bind')`

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;


const fzLocal = {
    my_demo_action: {
        cluster: 61453,
        type: 'raw',
        convert: (model, msg, publish, options, meta) => {
            const payload = {action: 'my_demo_action'};
            return payload;
        },
    },
};

const definition = {
    zigbeeModel: ['model.001'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: 'model.001', // Vendor model number, look on the device for a model number
    vendor: 'vendor001', // Vendor of the device (only used for documentation and startup logging)
    description: 'description001', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    fromZigbee: [fz.command_off, fz.command_on, fzLocal.my_demo_action, fz.battery], // We will add this later
    toZigbee: [], // Should be empty, unless device can be controlled (e.g. lights, switches).
    exposes: [e.battery()], // Defines what this device exposes, used for e.g. Home Assistant discovery and in the frontend
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
        await reporting.batteryPercentageRemaining(endpoint);
    },
};


module.exports = definition;
```

**Comment by Koenkk** [comment]
> Cannot read properties of undefined (reading 'bind')

Your code looks good but something goes wrong during ` await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);`, can you make sure you are running z2m 1.25.0? If you are, try putting `console.log(reporting)` above `await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);` and provide the log when configuring here.
